Reckless-Push/course_informer
# Full favorite support

# Pull Request (PR) Template

## Overview



This PR adds full support for favorite courses. Add favorite course buttons will add or remove a button from the current user's favorite courses. Courses on a user's favorites list will be highlighted yellow and will update when clicked.

## Changes



- Add favorite course buttons
- Add favorite courses in my profile

## How to Test



- Add a course to your favorites
- Check my profile for the course

## Screenshots



![Screenshot from 2024-01-01 22-20-36](https://github.com/Reckless-Push/course_informer/assets/111618981/439fd20e-68d5-4ed4-9a14-fe6780519247)
![Screenshot from 2024-01-01 22-20-42](https://github.com/Reckless-Push/course_informer/assets/111618981/3046d778-6adf-4087-a503-f09eef0168c2)

## Checklist

- [x] I have run the code locally and it does not produce any new warnings.
- [x] I have added comments to my code where necessary.

## Additional Notes
